server/datastore/mysql/hosts.go

```diff
@@ -1381,7 +1389,24 @@ func (ds *Datastore) updateOrInsert(ctx context.Context, updateQuery string, ins
 	return ctxerr.Wrap(ctx, err, "insert")
 }
 
-func (ds *Datastore) SetOrUpdateMunkiVersion(ctx context.Context, hostID uint, version string) error {
+func (ds *Datastore) SetOrUpdateMunkiInfo(ctx context.Context, hostID uint, version string, errors, warnings []string) error {
+	// NOTE(mna): if we lose too many saves due to some errors, we could continue
+	// processing even when part of the ingestion fails (e.g. process version if
+	// issues fail or vice-versa, ignore issues that failed to be created, etc.).
+	// Currently, taking a strict approach as a point that was mentioned in the
+	// ticket was to care about data accuracy, so instead of allowing to save
+	// only a subset of issues, we fail if we can't save the complete set.
+
+	// TODO(mna): figure out what we want to do regarding message length (truncate, accept longer, etc.)
```

Comment:
Awaiting a decision about it, was discussed a bit here: https://github.com/fleetdm/fleet/issues/6961#issuecomment-1216644458 and subsequent comments, and on slack: https://fleetdm.slack.com/archives/C03C41L5YEL/p1661274716041019?thread_ts=1661200559.241959&cid=C03C41L5YEL

Reply:
(that being said, once this gets approved, I'll merge so that the companion frontend work can progress, and will implement any changes required in a following PR).

Reply:
The decision is to go with 255 chars max, so since this is how the PR is implemented, I'll cleanup the TODO and ensure we truncate if we receive longer messages.